client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -74,13 +84,14 @@ const DomainPicker: FunctionComponent< Props > = ( {
 
 	const [ domainSearch, setDomainSearch ] = useState( siteTitle );
 
-	const placeholderAmount = 5;
+	const freePlaceholderAmount = 1;
+	const paidPlaceholderAmount = 5;
 
 	const [ search ] = useDebounce( domainSearch.trim() || defaultQuery || '', selectorDebounce );
 	const searchOptions = {
 		include_wordpressdotcom: true,
-		include_dotblogsubdomain: true,
-		quantity: 15,
+		include_dotblogsubdomain: false,
+		quantity: 6, // 1 free subdomain, 5 paid domains
```

Comment:
👍 Nice note

Do you think this quantity and the placeholder amounts should be related?

Reply:
It's up for debate.

I'm running into an edge cases (using `12` or `23` as keyword) where the API does not return free wordpress subdomain. The free subdomain UI gets stuck in placeholder animation forever. I believe the API should always return 1 free wordpress subdomain. I tried increasing the quantity in the API param, it doesn't help.

The only way to ensure that you get always 1 free subdomain and 5 paid domains is to have 2 API request. 1 API request specifically for free subdomain, and another API request specifically for paid domains.

Since there is going to be hook for the free subdomain coming, I intend to reuse that hook for the free subdomain, and update this API request specifically for only paid domains.

Reply:
>The only way to ensure that you get always 1 free subdomain and 5 paid domains is to have 2 API request. 1 API request specifically for free subdomain, and another API request specifically for paid domains.

I've understood two api separate requests is what we should indeed do: 

> If we want both subdomains and FQDN, we’ll want to do two separate queries.

via https://github.com/Automattic/wp-calypso/issues/40164

(didn't read the whole thread/PR so pardon if I missed something :-) )

cc @klimeryk

Reply:
@simison It's a long story. Here's how the story goes:

### Chapter 1

The initial problem was that 1 single API request could return 3 types of domain:
- **One** wordpress subdomain.
- **Multiple** dotblog subdomain.
- **Multiple** paid domains.

With this behaviour, sometimes there just too many dotblog subdomains and no paid domains in the API request. That was the initial problem.

### Chapter 2

When @klimeryk suggested that dotblog subdomains should not be displayed anymore, the solution to fix this problem was just to set `include_dotblogsubdomain=false`, which was the ideal solution, because it seems that the API request always returns:
- **One** wordpress subdomain.
- **Multiple** paid domains.

This PR is implemented based on this understanding, 1 API request to get everything exactly what we want.

### Chapter 3

However, I later discovered edge cases with keywords like `12` or `23` where the API request only returns:
- **MIssing** wordpress subdomain.
- **Multiple** paid domains.

And if I were to make another API request with edge case keywords like `12` or `23` specifically asking only for one wordpress subdomain using the param `only_wordpressdotcom=true`, then I'm able to get a wordpress subdomain for those edge case keywords.

Therefore, it comes back to the conclusion that, yes, we do need 2 API request afterall!

### Chapter 4

But instead of using 2 API request, I mentioned in my previous comment to:
- Use the `useFreeDomainSuggestion()` hook for the free subdomain (which is, at the end of the day, another API request).
- Modify the existing API request in this file to retrieve only paid domains.

This hook is `useFreeDomainSuggestion()` coming from this PR https://github.com/Automattic/wp-calypso/pull/40236/files#diff-1079e690e56573527885526f8fd10cd5.

The benefit of using this hook is to ensure that the free wordpress subdomain appears consistent throughout other places in the UI. Right now, there is a problem where the domain picker button may show subdomain, and the domain picker shows another subdomain.

Reply:
>The benefit of using this hook is to ensure that this subdomain appears consistent throughout other places in the UI. Right now, there is a problem where the domain picker button may show one subdomain, and the domain picker shows another subdomain, as mentioned in the known issues section in this PR description

Yeah that's not great and would be important to fix. 

Two digit keyword issues seems so edge-case that it doesn't matter for now.

We can also do all the API shuffling in separate PR, as this PR already improves the layout a lot.

Reply:
> where the API does not return free wordpress subdomain. The free subdomain UI gets stuck in placeholder animation forever.

This seems like a bug 🙂 Do we understand what's happening? Is there and error from the API?

Looking at this section and the one below it:

https://github.com/Automattic/wp-calypso/blob/0c337834ee20e98e151fc217056d16d0a031a065/client/landing/gutenboarding/components/domain-picker/index.tsx#L161-L173

`undefined` and `[]` will both render the placeholders. `undefined` means the resolver hasn't completed (or something is buggy with the data store code we should address), whereas `[]` means the result set is empty. Those are not the same condition and representing them both with placeholders isn't a good choice.

This is on the component end. On the other side, it's concerning to me that we don't return any WordPress.com subdomain results for some queries… that seems problematic.

When working on this initially, I noticed that the endpoint behaves differently with `include_wordpressdotcom` and `only_wordpressdotcom`. Try the same query with `only_wordpressdotcom`, and you'll get results. I don't know why 🤷‍♂

Reply:
> undefined and [] will both render the placeholders.

So how about `undefined` (loading) should render the placeholders, and `[]` should render some no results text hint?

> When working on this initially, I noticed that the endpoint behaves differently with `include_wordpressdotcom` and `only_wordpressdotcom`. Try the same query with `only_wordpressdotcom`, and you'll get results. 

Yup indeed, if using `only_wordpressdotcom` yields a wordpress subdomain, I'm sure they could have returned a wordpress subdomain even with edge case keywords on API requests without this param.